diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
index c575db7..796ba26 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
@@ -89,6 +90,9 @@ import com.reactnativecommunity.webview.events.TopRenderProcessGoneEvent;
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.io.UnsupportedEncodingException;
 import java.lang.IllegalArgumentException;
 import java.net.MalformedURLException;
@@ -916,6 +920,27 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
           webView.getId(),
           createWebViewEvent(webView, url)));
     }
+    @Override
+    public WebResourceResponse shouldInterceptRequest (WebView view,
+                                                       WebResourceRequest request) {
+      if (!request.getUrl().getScheme().contains("http"))
+        return null;
+
+      String host = request.getUrl().getHost();
+      if(host.contains("googletagmanager") || host.contains("google-analytics") || host.contains("sentry")) {
+        InputStream stream = new InputStream() {
+          @Override
+          public int read() throws IOException {
+            return -1;
+          }
+        };
+        WebResourceResponse resp = new WebResourceResponse("text/html", "utf-8", stream);
+        resp.setStatusCodeAndReasonPhrase(400, "Bad request");
+        return resp;
+      }
+
+      return null;
+    }
 
     @Override
     public boolean shouldOverrideUrlLoading(WebView view, String url) {
